### DISCUSSION TITLE
Windows builds are working!

### DISCUSSION THREAD

**Discussion by simonhamp** [NativePHP/laravel#271] [Announcements]
I couldn't hold myself back... I'm very excited!

## Windows builds are _almost_ here!
![](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExeDc5a2I2bDVpMWRkOWo2dTltMnc0YzB5dWs2aGJ6OHZxMGVyNG0wdSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/MGYY2FIs8VrfW/giphy-downsized.gif)

**That's right! 🎉 🔥**

I'm currently testing a set of Windows PHP builds with [sponsors](https://github.com/sponsors/simonhamp). We're now at a point where I believe static builds for Windows are working sufficiently well for development and testing purposes.

Windows is the number one issue for NativePHP right now. So I'm hoping to be able to release a new version of various parts of NativePHP that will support these binary executables in the coming _days_ and start working through that backlog of issues with a customary "please update and try again".

Hopefully, this will mean that the story of development on/for Windows will be much smoother. Assuming it works as well for everyone else as it has for me, this takes us **significantly closer to NativePHP working consistently across all 3 main platforms**.

It's a huge step for sure, but I also want to set expectations: there are still some unknowns right now. I don't use Windows for daily development and I have bumped into issues that you are likely not going to want end users of your apps to have to solve.

But in any case, it's still mega progress!

What I need from you—especially if you have a working Windows set up—is to test, test, and test some more, and report your findings in [issues](https://github.com/NativePHP/laravel/issues/new/choose).

Please be alert to duplicate issues—please use the search and check what others have posted before submitting an identical issue. 🙏

## More PHP versions

On top of this, this next release will include _multiple versions_ of PHP for **all platforms**:

- 8.1.28
- 8.2.18
- 8.3.6

Up to now, NativePHP has relied on one of the first builds we could generate statically for Mac & Linux. But this release will see us start rolling support for all of [the supported, stable releases of PHP](https://www.php.net/supported-versions.php) across the major platforms/architectures.

This will be a breaking change in the [`php-bin`](https://github.com/nativephp/php-bin) package, but it shouldn't require any downstream changes in your application.

All you will need to do is upgrade the version of [`nativephp/electron`](https://github.com/nativephp/electron) you're using. Details will be in the release notes of this package when the time comes.

## Thanks to Sponsors
I want to say a huge thanks to the sponsors who have been supporting me on NativePHP. Thanks to your support, I've been able to buy out the necessary time to get this work done.

Please go and give them a kind word on Discord, Twitter etc.

Obviously, with more support, the more time I can dedicate to getting things moving and the faster we can get there.

So if you've got any vested interest in NativePHP becoming production ready anytime soon, I kindly ask you to dig deep and support the project via [GitHub Sponsors](https://github.com/sponsors/simonhamp).

---

Thanks for your time. Have an awesome weekend!

**Comment by timoor97** [comment]
Good job thank you , We are waiting for Macos 👌

**Reply by simonhamp** [reply]
Mac support is already there, with PHP8.1 for now.

**Comment by JaberWiki** [comment]
That's great!

**Comment by mouradsme** [comment]
If I have a php.exe with pdo_mysql support, can I use MySQL instead of SQLite?
If yes, then kindly tell me how.
Thanks for your hard work

**Reply by simonhamp** [reply]
No, this isn't possible right now. We are working on a way for you to build PHP with your own extensions, but it won't be ready for a little while yet
